wire-kotlin-generator/src/main/java/com/squareup/wire/kotlin/KotlinGenerator.kt

```diff
@@ -1968,6 +1970,26 @@ class KotlinGenerator private constructor(
     }
   }
 
+  private fun addBuildClosure(type: MessageType, companionBuilder: TypeSpec.Builder) {
+    val body = buildCodeBlock {
+      addStatement("val builder = Builder()")
+      addStatement("buildFn(builder)")
+      addStatement("return builder.build()")
+    }
+    val buildFn = FunSpec.builder("build")
+      .addParameter("buildFn",
+        LambdaTypeName.get(
+          parameters = arrayOf(TypeVariableName("Builder")),
+          returnType = Unit::class.asClassName()
+        )
```

Comment:
Probably more ergonomic to change the builder type to be the receiver rather than an argument.

Also please pass in the `builderClassName` from the calling function and use that to reference the type here...

Reply:
See comment below on using builder as the receiver rather than the argument.

👍  on builderClassName